src/script/common/c_content.cpp

```diff
@@ -246,12 +246,35 @@ TileDef read_tiledef(lua_State *L, int index)
 }
 
 /******************************************************************************/
-ContentFeatures read_content_features(lua_State *L, int index)
+void push_tiledef(lua_State *L, TileDef tiledef)
```

Comment:
Wait

Isn't TileDef a really huge structure?  Why are you passing it by value?